src/wallet/walletdb.cpp

```diff
@@ -715,6 +708,45 @@ bool ReadKeyValue(CWallet* pwallet, CDataStream& ssKey, CDataStream& ssValue, CW
                 return false;
             }
         }
+        else if (strType == "hdchain")
+        {
+            CHDChain chain;
+            ssValue >> chain;
+            if (!pwallet->SetHDChain(chain, true))
+            {
+                strErr = "Error reading wallet database: SetHDChain failed";
+                return false;
+            }
+        }
+        else if (strType == "chdchain")
```

Comment:
This block is exactly equal to the block above.

Reply:
Will clean it up shortly

Reply:
Cleaned up that code.Closing this and marking as resolved